app/users/routes.py

```diff
@@ -0,0 +1,290 @@
+    # check if stored hashed password matches current password in form
+    if check_password_hash(
+            user["user_password"], request.form.get("current_password")):
+        flash(Markup(
+            f"<i class='far fa-check-circle green-text'></i>\
+            Your password has been updated successfully!"))
+        users_collection.update_one(
+            {"username_lower": session["user"].lower()},
+            {"$set": {"user_password": generate_password_hash(
+                request.form.get("new_password"))}})
+    else:
+        # password does not match
+        flash(Markup(
+            f"<i class='fas fa-exclamation-circle red-text'></i>\
+            Whoops! Looks like your <span class='pink-text text-lighten-2'>\
+            password</span> is incorrect. Please try again."))
+    # redirect back to profile
+    return redirect(url_for("users.profile", username=username))
+
+
+# ----- DELETE ACCOUNT ----- #
+@users.route("/profile/<username>/delete", methods=["GET", "POST"])
+def profile_delete_account(username):
+    """
+    Delete entire user account if desired.
+
+    Each user has the option to delete their account.
+    This would delete all of their recipes, too.
+    """
+    user = get_user_lower(session["user"])
+    # check if stored hashed password matches current password in form
+    if check_password_hash(
+            user["user_password"], request.form.get("verify_password")):
+        # find all recipes belonging to user
+        user_recipes = [recipe for recipe in user.get("user_recipes")]
+        # for each user recipe, remove from database and pull from others' favs
+        for recipe in user_recipes:
+            recipes_collection.remove({"_id": recipe})
+            users_collection.update_many({}, {"$pull": {"user_favs": recipe}})
+        # find all recipes that the user saved
+        user_favs = [recipe for recipe in user.get("user_favs")]
+        # decrement each recipe the user liked by -1
+        for recipe in user_favs:
+            recipes_collection.update_one(
+                {"_id": recipe}, {"$inc": {"user_favs": -1}})
+        flash(Markup(f"<i class='fas fa-user-times red-text'></i>\
+            Your account and recipes have been successfully deleted."))
+        # remove user from session cookies and delete user from database
+        session.pop("user")
+        users_collection.remove({"_id": user.get("_id")})
+        return redirect(url_for("main.home"))
+
+    else:
+        # password does not match, so redirect back to profile
+        flash(Markup(f"<i class='fas fa-exclamation-circle red-text'></i>\
+            Whoops! Looks like your <span class='pink-text text-lighten-2'>\
+            password</span> is incorrect. Please try again."))
+        return redirect(url_for("users.profile", username=username))
+
+
+# ----- ADMIN DELETE USERS ----- #
+@users.route("/admin/<username>/delete", methods=["GET", "POST"])
+def admin_delete_user(username):

```

Comment:
Remember to write DRY code. Some of the code you have in the route above is exactly same with what you have in this route. Try to put those code in a function and call the function from both routes.